hca/util/fs_helper.py

```diff
@@ -0,0 +1,90 @@
+import os, tempfile, errno
+from datetime import datetime
+
+
+class FSHelper(object):
+    """
+    Static helper class providing convenience methods for reading and writing files.
+    """
+
+    @staticmethod
+    def atomic_write(filename, shared_dir, content):
+        """
+        Atomically writes content a to file.
+
+        Writes to a temp file and renames file (atomic operation) to desired filename.
+        To ensure atomicity, the src and dest dirs must be the same.
+
+        :param filename: File name only (e.g. example.txt)
+        :param shared_dir: Shared directory of temp and dest file
+        :param content: Content to write to file
+        """
+        temp_filename = ""
+        dest_filename = os.path.join(shared_dir, filename)
+        try:
+            with tempfile.NamedTemporaryFile('wb', dir=shared_dir, delete=False) as temp:
+                temp_filename = temp.name
+                temp.write(content)
+            FSHelper._atomic_rename(temp_filename, dest_filename)
+        finally:
+            if temp_filename and os.path.isfile(temp_filename):
+                try:
+                    os.remove(temp_filename)
+                except EnvironmentError:
+                    print("Failed to clean up temporary file {}".format(temp_filename))
+                    raise
```

Comment:
I don't think this is a case where we should fatal.  I would just warn and move on.

Reply:
Makes sense